File: 4th semester/Systems for design and implementation/lab11WebApp/src/app/spell/shared/spell.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { map, Observable } from 'rxjs';
import { Spell } from './spell.model';

@Injectable({
  providedIn: 'root'
})
export class SpellService {

  private url = "http://localhost:8080/api/spells"

  constructor(private http: HttpClient) { }

  getSpells(): Observable<Spell[]> {
    return this.http.get<Array<Spell>>(this.url).pipe(map(res => {return res['spells']}));
  }

  getSpell(id: number): Observable<Spell> {
    return this.getSpells().pipe(map(spells => spells.find(Spell => Spell.id === id)))
  }

  saveSpell(Spell: Spell): Observable<Spell> {
    console.log("save Spell", Spell)

    return this.http.post<Spell>(this.url, Spell)
  }

  updateSpell(Spell: Spell): Observable<Spell> {
    return this.http.put<Spell>(this.url + "/" + Spell.id, Spell)
  }

  deleteSpell(id: number): Observable<any> {
    return this.http.delete(this.url + "/" + id)
  }

  filterSpells(name: string) {
    return this.http.get<Array<Spell>>(this.url + "/filter/" + name).pipe(map(res => {return res['spells']}));
  }

  sortSpells() {
    return this.http.get<Array<Spell>>(this.url + "/sort").pipe(map(res => {return res['spells']}));
  }
}

File: 4th semester/Systems for design and implementation/lab10/SocketsProject/common/src/main/java/Service/SpellService.java
package Service;

import Model.Spell;
import Model.ValidatorException;

import java.util.List;
import java.util.concurrent.Future;

public interface SpellService {
    String ADD = "addSpell";
    String DELETE = "deleteSpell";
    String UPDATE = "updateSpell";
    String GET = "getSpells";

    Future<Spell> addSpell(Long id, String name, String description) throws ValidatorException;
    Future<Spell> deleteSpell(Long id);
    Future<Spell> updateSpell(Long id, String name, String description);
    Future<List<Spell>> getSpells();
}
